files/plugins/check_cis_audit.py

```diff
@@ -149,13 +153,7 @@ def parse_args(args):
     parser.add_argument(
         "--cis-profile",
         "-p",
-        choices=[
-            "",
-            "level1_server",
-            "level2_server",
-            "level1_workstation",
-            "level2_workstation",
-        ],
+        choices=PROFILE_OPTIONS + [""],
```

Comment:
I think this will break backward compatibility?  Will users need to change the `cis_audit_profile` config option after upgrading?

Can we retain backward compatibility by accepting the same options (same config.yaml, same choices here), but mapping it after parsing the arg based on the distro version and PROFILE_MAP?

Reply:
I don't know how no one complained about yet, but right now the cis audit is broken right now for Ubuntu series > Focal.

The choices are fixed to the bionic option, e.g: `level1_server` (without the `cis_` prefix) and this configure the cron job without mapping to the new profiles:

https://github.com/canonical/charm-nrpe/blob/21f1a23382ad90b4a0d5e67e4105c3bc534108bd/hooks/nrpe_utils.py#L236-L241

 So, in a Focal environment passing `level1_server` the cronjob calls the binary and will fail saying that  the option passed is not valid. And if you try to pass `cis_level1_server` the cronjob will work, but the check_cis_audit will say that it's not a valid option.
 
 So I don't think we have a compatibility issue after this change, it will just let it run normally

Reply:
No I mean we can retain backward compatibility - consider the case where one is running this on bionic with it working with `level1_server` in the config options.  Then they upgrade to focal and it breaks because of this bug.  We should fix the bug so the user can refresh the charm and get the fix.  With your patch here, the user would have to then also change the config option.  If we retain backward compatibility they won't have to.

> The choices are fixed to the bionic option, e.g: level1_server (without the cis_ prefix) and this configure the cron job without mapping to the new profiles:

Can we retain the choices (`level1_server`, etc.), but do the mapping before configuring the cronjob.  So the config options stay the same, but the options are mapped within the charm.

Reply:
+1 for retain backward compatibility.

Reply:
In that case I need #188 to be merged and change the logic at `cis_cmd_params`. Let's finish there and then I rebase and continue this work